code/modules/library/library_computer.dm

```diff
@@ -0,0 +1,577 @@
+
+	static_data["programmatic_booklist"] = list()
+	for(var/book in GLOB.library_catalog.books)
+		var/datum/programmatic_book/PB = book
+		var/list/book_data = list(
+			"title" = PB.title  ? PB.title : "not specified",
+			"author "= PB.author ? PB.author : "Nanotrasen",
+			"id" = PB.id,
+		)
+		static_data["programmatic_booklist"] += list(book_data)
+
+	return static_data
+
+/obj/machinery/computer/library/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if(..())
+		return
+
+	if(ui_act_modal(action, params))
+		return
+
+	switch(action)
+		//Page Switching
+		if("incrementpage")
+			archive_page_num = clamp(archive_page_num + 1, 1, num_pages)
+			populate_booklist()
+		if("incrementpagemax")
+			archive_page_num = num_pages
+			populate_booklist()
+		if("deincrementpage")
+			archive_page_num = clamp(archive_page_num - 1, 1, num_pages)
+			populate_booklist()
+		if("deincrementpagemax")
+			archive_page_num = 1
+			populate_booklist()
+		//Search Tools' Buttons
+		if("toggle_search_category")
+			var/category_id = text2num(params["category_id"])
+			if(category_id in user_data.search_categories)
+				user_data.search_categories -= category_id
+				populate_booklist()
+			else
+				user_data.search_categories += category_id
+				populate_booklist()
+		if("clear_search")
+			user_data.clear_search()
+			populate_booklist()
+		if("find_users_books")
+			user_data.clear_search() //we need to clear out other search params first
+			user_data.search_ckey = params["user_ckey"]
+			populate_booklist()
+		if("clear_ckey_search")
+			user_data.search_ckey = null
+			populate_booklist()
+
+		//Order Buttons
+		if("order_external_book")
+			var/datum/cachedbook/orderedbook = GLOB.library_catalog.get_book_by_id(params["bookid"])
+			if(orderedbook && print_cooldown <= world.time)
+				make_external_book(orderedbook)
+				print_cooldown = world.time + PRINTING_COOLDOWN
+		if("order_programmatic_book")
+			var/datum/programmatic_book/PB = GLOB.library_catalog.get_programmatic_book_by_id(params["bookid"])
+			if(PB && print_cooldown <= world.time)
+				make_programmatic_book(PB)
+				print_cooldown = world.time + PRINTING_COOLDOWN
+		//book author actions
+		if("delete_book")
+			if(params["bookid"])
+				var/datum/cachedbook/selectedbook = GLOB.library_catalog.get_book_by_id(params["bookid"])
+				if(!selectedbook)
+					playsound(src, 'sound/machines/synth_no.ogg', 15, TRUE)
+					atom_say("Deletion Failed!")
+				if(selectedbook.ckey != params["user_ckey"])
+					message_admins("[params["user_ckey"]] attempted to delete a book that wasn't theirs, this shouldn't happen, please investigate.")
+				if(GLOB.library_catalog.remove_book_by_id(params["bookid"])) //this doesn't need to be logged
+					playsound(loc, 'sound/machines/ping.ogg', 25, 0)
+					atom_say("Deletion Succesful!")
+				else
+					playsound(src, 'sound/machines/synth_no.ogg', 15, TRUE)
+					atom_say("Deletion Failed!")
```

Comment:
```suggestion
					atom_say("Deletion Succesful!")
					return
				playsound(src, 'sound/machines/synth_no.ogg', 15, TRUE)
				atom_say("Deletion Failed!")
```
Goes for the other stuff below as well.
For this one you could switch it around. So the "default" case is that it succeeds

Reply:
done